internal/common/tasks/scripts/push_artifact.sh

```
@@ -1,6 +1,18 @@
 #!/bin/sh
 set -e
 
+# Format: stage|done|total (pipe-delimited, stored as a pod annotation)
+emit_progress() {
+  _ep_stage="$1"; _ep_done="$2"; _ep_total="$3"
+  curl -s --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+    -X PATCH \
+    -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    -H "Content-Type: application/merge-patch+json" \
+    "https://${KUBERNETES_SERVICE_HOST}:${KUBERNETES_SERVICE_PORT}/api/v1/namespaces/$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)/pods/${HOSTNAME}" \
```

Shellcheck (0.11.0)

[warning] In POSIX sh, HOSTNAME is undefined.

(SC3028)